app/src/main/java/com/tengu/sharetoclipboard/ShareToClipboardActivity.java

```diff
@@ -33,15 +33,11 @@ protected void onCreate(Bundle savedInstanceState) {
         Intent intent = getIntent();
         String action = intent.getAction();
         String type = intent.getType();
+
         if (!Intent.ACTION_SEND.equals(action) || type == null) {
             finish();
             return;
         }
-        if (!type.startsWith("text/")) {
```

Comment:
Can we extract the switch case to a method and change this to:

if (type.startsWith("text/")) {
   ...
} else if type.startsWith("text/")) {
   handleSendImage(intent);
} else {
  handleSendText(intent, R.string.error_type_not_supported_by_platform);
}